server/src/modules/intentions/queries/findDemandes.query.ts

```diff
@@ -16,8 +21,75 @@ export const findDemandes = async ({
 }) => {
   const demandes = await kdb
     .selectFrom("demande")
-    .selectAll()
+    .leftJoin("dataFormation", "dataFormation.cfd", "demande.cfd")
+    .leftJoin("dataEtablissement", "dataEtablissement.uai", "demande.uai")
+    .leftJoin("dispositif", "dispositif.codeDispositif", "demande.dispositifId")
+    .selectAll("demande")
+    .select("dataFormation.libelle as libelleDiplome")
+    .select("dataEtablissement.libelle as libelleEtablissement")
+    .select("dispositif.libelleDispositif as libelleDispositif")
     .select(sql<string>`count(*) over()`.as("count"))
+    .select((eb) =>
+      sql<boolean>`${eb.exists((eb) =>
+        eb
+          .selectFrom(["demande as demandeCompensee"])
+          .where("demande.typeDemande", "in", [
+            "augmentation_compensation",
+            "ouverture_compensation",
+          ])
+          .whereRef("demandeCompensee.cfd", "=", "demande.compensationCfd")
+          .whereRef("demandeCompensee.uai", "=", "demande.compensationUai")
+          .whereRef(
+            "demandeCompensee.dispositifId",
+            "=",
+            "demande.compensationDispositifId"
+          )
+          .selectAll()
+      )}`.as("estCompensee")
+    )
```

Comment:
Je vois pas pourquoi on en a besoin de vérifier le type, si la demande est compensée alors on aura un compensationCfd.

Autre syntaxe:
```suggestion
    .select(({ exists }) =>
      exists((eb) =>
        eb
          .selectFrom(["demande as demandeCompensee"])
          .where("demande.typeDemande", "in", [
            "augmentation_compensation",
            "ouverture_compensation",
          ])
          .whereRef("demandeCompensee.cfd", "=", "demande.compensationCfd")
          .whereRef("demandeCompensee.uai", "=", "demande.compensationUai")
          .whereRef(
            "demandeCompensee.dispositifId",
            "=",
            "demande.compensationDispositifId"
          )
          .selectAll()
      ).as("estCompensee")
    )
```